### PR TITLE
Adding a restart test to make sure #16887 doesnt happens again

### PR DESCRIPTION
**Pull request by mavenugo**
Thanks to @cpuguy83 for the test script and @jfrazelle for `but, I wont forget` :smile: 

Signed-off-by: Madhu Venugopal madhu@docker.com

### CODE REVIEW

**Comment by jfrazelle** [comment]
hahhaa awesome :)

**Comment by jfrazelle** [comment]
LGTM

File: integration-cli/docker_cli_restart_test.go [hunk 1]
```diff
@@ -148,3 +148,22 @@ func (s *DockerSuite) TestContainerRestartwithGoodContainer(c *check.C) {
...
+func (s *DockerSuite) TestContainerRestartSuccess(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	out, _ := dockerCmd(c, "run", "-d", "--restart=always", "busybox", "top")
+	id := strings.TrimSpace(out)
+	c.Assert(waitRun(id), check.IsNil)
+
+	pid, err := inspectField(id, "State.Pid")
+	c.Assert(err, check.IsNil)
+	dockerCmd(c, "run", "--pid=host", "busybox", "kill", "-9", pid)
```
Comment (jfrazelle) [inline integration-cli/docker_cli_restart_test.go]: pid host needs `NotUserNamespace` :P

**Comment by cpuguy83** [comment]
LGTM

File: integration-cli/docker_cli_restart_test.go (hunk 1 above)
Comment (cpuguy83) [inline integration-cli/docker_cli_restart_test.go]: Could also use syscall, but I figure containerize even my kills.

Comment (cpuguy83) [inline integration-cli/docker_cli_restart_test.go]: Probably good to make sure this runs on userns.

Comment (jfrazelle) [inline integration-cli/docker_cli_restart_test.go]: good idea :)

**Comment by thaJeztah** [comment]
fails on userns;

```
FAIL: docker_cli_restart_test.go:128: DockerSuite.TestContainerRestartSuccess

docker_cli_restart_test.go:137:
    dockerCmd(c, "run", "--pid=host", "busybox", "kill", "-9", pid)
/go/src/github.com/docker/docker/pkg/integration/dockerCmd_utils.go:35:
    c.Assert(err, check.IsNil, check.Commentf("%q failed with errors: %s, %v", strings.Join(args, " "), out, err))
... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc2082c1760)} ("exit status 1")
... "run --pid=host busybox kill -9 26074" failed with errors: Timestamp: 2015-10-18 22:46:03.224681147 +0000 UTC
Code: System error

Message: operation not permitted
```

**Comment by tiborvass** [comment]
Ping @mavenugo for https://github.com/docker/docker/pull/17005#discussion_r42071818

**Comment by cpuguy83** [comment]
Ping, really need this test in.

**Comment by cpuguy83** [comment]
Let me know if you want me to carry.

**Comment by thaJeztah** [comment]
Relates to https://github.com/docker/docker/issues/16887

**Comment by LK4D4** [comment]
@cpuguy83 pls carry if you have time.

**Comment by mavenugo** [comment]
apologies on the delay guys. will get it done today.

**Comment by LK4D4** [comment]
@mavenugo Oh, cool, thanks!

**Comment by LK4D4** [comment]
@mavenugo Did you have a chance to look at this?

**Comment by mavenugo** [comment]
@cpuguy83 @jfrazelle finally got around to this. Used native golang calls to ungracefully kill the container which should work for userns case as well. Also, the RestartCount test requires `waitInspect` to account for monitor to kick in to do its job of restarting the container.

**Comment by vdemeester** [comment]
:+1: 

LGTM 🐼

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**